src/main/java/Main.java

```diff
@@ -1,217 +1,144 @@
-
-        HashMap<String, List<Document>> map = new HashMap<>();
-
-        FeedOptions queryOptions = new FeedOptions();
-
-        // note that setMaxItemCount sets the number of items to return in a single page result
-        queryOptions.setMaxItemCount(pageSize);
-        queryOptions.setEnableCrossPartitionQuery(true);
-        queryOptions.setRequestContinuation(continuationToken);
+    private void queryDocumentsByPage() {
+        CosmosQueryRequestOptions queryOptions = new CosmosQueryRequestOptions();
 
         String collectionName = "volcanoCollection";
-        String sql = "SELECT * FROM volcanoCollection";
-
-        String collectionLink = String.format("/dbs/%s/colls/%s", databaseName, collectionName);
-        Observable<FeedResponse<Document>> queryObservable =
-                client.queryDocuments(collectionLink,
-                        sql, queryOptions);
-
-        //Observable to Interator
-        Iterator<FeedResponse<Document>> it = queryObservable.toBlocking().getIterator();
-
-        FeedResponse<Document> page = it.next();
-        List<Document> results = page.getResults();
-        for (Document doc : results) {
-            JSONObject obj = new JSONObject(doc.toJson());
-            String id = obj.getString("id");
-
+        String containerSql = String.format("SELECT * from c where c.id = '%s'", collectionName);
+        System.out.println(containerSql);
+        String itemSql = String.format("SELECT * FROM %s", collectionName);
+        System.out.println(itemSql);
+
+        CosmosPagedIterable<CosmosContainerProperties> results = database.queryContainers(containerSql, queryOptions);
+
+        // Iterating through all the results.
+        // If there are more than a single page of results, it will auto-magically fetch the next page.
+        for (CosmosContainerProperties cosmosContainerProperties : results) {
+            String id = cosmosContainerProperties.getId();
+            CosmosContainer container = database.getContainer(id);
+            container.queryItems(itemSql, queryOptions, TestObject.class).forEach(item -> {
+                System.out.println(item.toString());
+            });
+            System.out.println(id);
         }
-        continuationToken = page.getResponseContinuation();
-        System.out.println("continuationToken2: " + continuationToken);
-        map.put(continuationToken, results);
-        return map;
     }
 
-
-
-    private void executeSimpleQueryWithList() throws JSONException {
-
-        FeedOptions queryOptions = new FeedOptions();
-        // note that setMaxItemCount sets the number of items to return in a single page result
-        queryOptions.setMaxItemCount(500);
-        queryOptions.setEnableCrossPartitionQuery(true);
-
-        HashMap<Integer, List<JSONObject>> docsPerPage = new HashMap<>();
+    private void executeSimpleQueryWithList(){
```

Comment:
Did you format this document? Alt+Shift+L on Windows. There should be a space after this.